crates/uv-pep508/src/marker/tree.rs

```diff
@@ -493,6 +494,19 @@ pub enum MarkerExpression {
     },
 }
 
+/// The kind of a [`MarkerExpression`].
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, PartialOrd, Ord)]
+pub(crate) enum MarkerExpressionKind {
```

Comment:
Is this _only_ used for backcompat sorting? If so, it might be useful to add a comment indicating as such. And that the order here matters?